src/firmware/app/control/trajectory_planner.c

```diff
@@ -195,17 +78,34 @@ app_trajectory_planner_interpolateConstantPeriodPositionTrajectory(
                            variable_time_trajectory->trajectory_elements[i - 1].time) +
                 variable_time_trajectory->trajectory_elements[i - 1].position.y;
 
-            const float interpolated_speed =
-                slope_speed *
+            const float interpolated_theta =
+                slope_theta *
+                    (trajectory_time -
+                     variable_time_trajectory->trajectory_elements[i - 1].time) +
+                variable_time_trajectory->trajectory_elements[i - 1].orientation;
+
+            const float interpolated_angular_speed =
+                slope_angular_speed *
                     (trajectory_time -
                      variable_time_trajectory->trajectory_elements[i - 1].time) +
-                variable_time_trajectory->speed_profile[i - 1];
+                variable_time_trajectory->trajectory_elements[i - 1].angular_speed;
+
+            const float interpolated_linear_speed =
+                slope_linear_speed *
+                    (trajectory_time -
+                     variable_time_trajectory->trajectory_elements[i - 1].time) +
+                variable_time_trajectory->trajectory_elements[i - 1].linear_speed;
```

Comment:
This might be a bit of a contentious idea, but a macro here might help shorten things up:
```
# define INTERPOLATE(FIELD_NAME) \
const float interpolated_##FIELD_NAME = \
    slope_##FIELD_NAME * \
    (trajectory_time - 
    variable_time_trajectory->trajectory_elements[i - 1].time) +